src/StateManagers/CoordinatesManager.vala

```diff
@@ -105,63 +108,68 @@ public class Akira.StateManagers.CoordinatesManager : Object {
     }
 
     /**
-     * Reset the coordinates to get the newly updated coordinates from the item.
-     * The coordinates change came from a canvas action that already moved the items
-     * therefore we set the d_update to false to prevent updating  the selected
-     * items Cairo transform.
+     * Reset the coordinates to get the newly updated coordinates from the selected items.
+     * This method is called when items are moved from the canvas, so we only need to update
+     * the X and Y values for the Transform Panel without triggering the update_items_*().
      */
-    private void on_reset_state_coords (Lib.Items.CanvasItem item) {
-        do_update = false;
+    private void on_reset_state_coords () {
+        on_init_state_coords ();
```

Comment:
yes! let's remove all cases where an event takes items...it's asking for trouble!